pkg/engine/health/health.go

```diff
@@ -16,6 +16,26 @@ import (
 	kudov1beta1 "github.com/kudobuilder/kudo/pkg/apis/kudo/v1beta1"
 )
 
+// IsTerminallyFailed returns whether an object is in a terminal failed state and has no chance to reach healthy
+func IsTerminallyFailed(obj runtime.Object) bool {
+	if obj == nil {
+		return false
+	}
+
+	switch obj := obj.(type) {
+	case *batchv1.Job:
+		if obj.Spec.BackoffLimit != nil {
+			if obj.Status.Failed > *obj.Spec.BackoffLimit {
```

Comment:
Would that be enough? According to https://kubernetes.io/docs/concepts/workloads/controllers/jobs-run-to-completion/#job-termination-and-cleanup, there could also be a `activeDeadlineSeconds` configured. Looks like we should check that the Jobs status is of `type: Failed` with `reason: DeadlineExceeded`. WDYT?

Reply:
@nfnt Good point, I changed that to use the Conditions.